actionview/lib/action_view/helpers/translation_helper.rb

```diff
@@ -94,7 +94,9 @@ def translate(key, **options)
             break translated unless translated.equal?(MISSING_TRANSLATION)
           end
 
-          break alternatives.first if alternatives.present? && !alternatives.first.is_a?(Symbol)
+          if alternatives.present? && !alternatives.first.is_a?(Symbol)
```

Comment:
Does ensuring that `alternatives` exists as an Array give us an opportunity to re-arrange this conditional?

```ruby
alternatives = Array.wrap(options.delete(:default) || []).compact

# ...
case alternatives.first
when NilClass # maybe `when nil`?
  break false
when Symbol
  key = alternatives.shift
else
  break I18n.translate(**options, default: alternatives)
end
```

Reply:
Avoiding that array allocation was initially done in #32747, so I wanted to preserve that behavior.

Also, however, `translate(:missing, default: nil)` should return `nil`, while `translate(:missing, default: [])` should return a missing translation error.  To support these two cases, the `alternatives.present?` check differentiates between `[nil]` and `[]`.  Using `case alternatives.first` would only handle the former.

(And to make things more complicated, `translate(:missing, default: [nil])` should return a translation missing error, to mirror `I18n.translate`.  Hence the `compact` on line 74 and the `alternatives.first && ...` on line 98.)